src/main/java/seedu/address/model/person/Availability.java

```diff
@@ -0,0 +1,54 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Candidate's availability.
+ * Guarantees: immutable; is valid as declared in {@link #isValidDate(String)}
+ */
+public class Availability {
+
+    public static final String MESSAGE_CONSTRAINTS = "Availability should consists of a list of numbers from 1 to 7, "
+            + "separated by commas ','";
+    public static final String VALIDATION_REGEX = "^[1-7]{1}+(?:,[1-7]{1}+)*$";
+
+    public final String availability;
+
+    /**
+     * Constructs a {@code Availability}.
+     *
+     * @param availability A valid available date.
+     */
+    public Availability(String availability) {
+        requireNonNull(availability);
+        checkArgument(isValidDate(availability), MESSAGE_CONSTRAINTS);
+        this.availability = availability;
+    }
+
+    /**
+     * Returns true if a given string is a valid date format.
+     */
+    public static boolean isValidDate(String test) {
```

Comment:
Just a tiny nitpick, maybe could use `isValidDay` instead of `isValidDate` in subsequent PRs, since we are specifying day instead of date - also, I am likely to use `isValidDate` for scheduling 😝

Reply:
Day sounds more in line with what we want, I will keep that in mind 😄

Reply:
+1! For v1.3. Let's KIV!